kbs/src/policy_engine/mod.rs

```diff
@@ -83,16 +58,30 @@ pub(crate) struct PolicyEngine(pub Arc<Mutex<dyn PolicyEngineInterface>>);
 
 impl PolicyEngine {
     /// Create and initialize PolicyEngine
-    pub async fn new(config: &PolicyEngineConfig) -> Result<Self, ResourcePolicyError> {
-        let policy_engine: Arc<Mutex<dyn PolicyEngineInterface>> = {
-            cfg_if::cfg_if! {
-                if #[cfg(feature = "opa")] {
-                    Arc::new(Mutex::new(opa::Opa::new(config.policy_path.clone().unwrap_or(PathBuf::from(DEFAULT_POLICY_PATH)))?))
-                } else {
-                    compile_error!("Please enable at least one of the following features: `opa` to continue.");
-                }
-            }
-        };
+    pub async fn new(config: &PolicyEngineConfig) -> Result<Self> {
+        let policy_engine: Arc<Mutex<dyn PolicyEngineInterface>> =
+            Arc::new(Mutex::new(opa::Opa::new(config.policy_path.clone())?));
         Ok(Self(policy_engine))
     }
+
+    pub async fn evaluate(&self, accessed_path: String, input_claims: String) -> Result<bool> {
+        self.0
+            .lock()
+            .await
+            .evaluate(accessed_path, input_claims)
+            .await
+    }
+
+    pub async fn set_policy(&self, request: &[u8]) -> Result<()> {
+        let request: Value = serde_json::from_slice(request)
+            .map_err(|_| PolicyEngineError::IllegalSetPolicyRequest)?;
+        let policy = request
+            .pointer("/policy")
+            .ok_or(PolicyEngineError::IllegalSetPolicyRequest)?
+            .as_str()
+            .ok_or(PolicyEngineError::IllegalSetPolicyRequest)?
+            .to_string();
```

Comment:
see comment on `PolicyEngine::set_policy()` signature, ideally we can just pass the str reference